salesagility/SuiteCRM
# Feature/api client admin panel

Updates and improves the OAuth2Clients and OAuth2Tokens admin panel.
Adds Client Credentials grant type.

## Description




- Unified OAuth2Tokens and OAuth2Clients into the same admin menu item.
- Fixes missing labels and icons.
- Adds a an allowed grant type to each client.
- Enables Client Credentials grant type on the OAuth2 server.
- Adds a default user to Client Credentials grant type, definable in each OAuth2Client.
- Adds a custom edit and detail view for each allowed grant type for OAuth2Client.
- Show related client in OAuth2Tokens list view
- Show related user in OAuth2Tokens list view
- Add options to filter OAuth2Tokens by client, user, grant type, active etc
- Add 'Active OAuth2Tokens' subpanel to OAuth2Client
- Provide a default sample data SQL dump for API testing

## Motivation and Context

Makes the OAuth2Clients and OAuth2Tokens modules more user friendly.
Adds Client Credentials grant type.

## How To Test This

Go to OAuth2Clients and OAuth2Tokens admin menu item
Create OAuth2Clients of different types
Create OAuth2Tokens through any type of client or by running the API testsuite
Ensure the filtering and works and the tokens link correctly to Users and OAuth2Clients

## Types of changes


### Final checklist
